src/tools/rbd/action/Bench.cc

```diff
@@ -103,6 +106,20 @@ void validate(boost::any& v, const std::vector<std::string>& values,
     v = boost::any(io_type);
 }
 
+void validate(boost::any& v, const std::vector<std::string>& values,
+              PatternByte *target_type, int) {
+  po::validators::check_first_occurrence(v);
+  const std::string &s = po::validators::get_single_string(values);
+  uint16_t pattern;
+  if (s == "random")
+    pattern = PATTERN_BYTE_RANDOM;
+  else if (s.size() == 1)
+    pattern = uint16_t(s.front());
```

Comment:
Accepting only a single character seems very restrictive -- this makes it next to impossible to specify a non-printable or an extended ASCII byte.  I'd do something like:
```
const int PATTERN_BYTE_RAND = -1;
const int PATTERN_BYTE_RAND_STR = -2;

...

void validate(boost::any& v, const std::vector<std::string>& values,
              PatternByte *target_type, int) {
  po::validators::check_first_occurrence(v);
  const std::string &s = po::validators::get_single_string(values);
  if (s == "rand") {
    v = boost::any(PATTERN_BYTE_RAND);
  } else if (s == "rand-str") {
    v = boost::any(PATTERN_BYTE_RAND_STR);
  } else {
    try {
      int pattern_byte = boost::lexical_cast<uint32_t>(s);
      if (pattern_byte >= 0 && pattern_byte <= 255) {
        v = boost::any(pattern_byte);
        return;
      }
    } catch (const boost::bad_lexical_cast &) {
    }
    throw po::validation_error(po::validation_error::invalid_option_value);
  }
}
```
(switching `pattern_byte` to `int` throughout)